### ISSUE TITLE
Get NWBE docker from OSBv2 building in tests

### ISSUE THREAD

**Issue by pgleeson** [OpenSourceBrain/DANDIArchiveShowcase#28]
@Daksh1603 building the NWBE docker image is working locally, but failing in the tests. See https://github.com/OpenSourceBrain/DANDIArchiveShowcase/actions/runs/5124779382/jobs/9217081351

Can you have a look?

**Comment by Daksh1603** [comment]
Yes, I'll work on this .

**Comment by sanjayankur31** [comment]
We looked at it during the meeting today. Daksh had made some tweaks and opened a PR in the OSBv2 repo. I suggested copying the docker file in here for the moment and making changes to get it to work---once solved, we can look at whether these changes are appropriate for the OSBv2 repo.

@Daksh1603 : I'd start by running this to see if it works:


```
 git clone https://github.com/OpenSourceBrain/OSBv2
 cd OSBv2/applications/nwb-explorer
 docker build -t mynwbosb .
```

I just tried it and also got the same error as CI:

```
unable to convert uid/gid chown string to host mapping: can't find gid for group jovyan: no such group: jovyan
```

I had a quick look and managed to diagnose it. The error seems to be related to using `NB_UID=jovyan` instead of `NB_USER=jovyan` here: https://github.com/OpenSourceBrain/OSBv2/blob/master/applications/nwb-explorer/Dockerfile#L27

The `NB_UID` field is meant to be numeric as per the jupyter docker docs:

https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#user-related-configurations

So, just making that one change fixes my local build. Not sure why it works on OSBv2---different docker versions?

So,  @Daksh1603 : things to do here:

- reproduce the error on your machine
- see if my fix (replacing `NB_UID` with `NB_USER`) fixes it
- tweak the CI file to also print the version of docker (new step that goes: `docker version`). My system gives: `Docker version 24.0.2, build cb74dfc`.
- if this fix works, I guess we should make it in the OSBv2 repo too: we can discuss that in the next meeting etc.

How does that sound?
